src/main/apiClient.ts

```diff
@@ -94,12 +94,14 @@ type SingleFactorPasswordlessParams = {
   phoneNumber?: string
 }
 
+type SingleFactorPasswordlessWithCaptcha = SingleFactorPasswordlessParams & {  captchaToken?: string }
+
```

Comment:
Why not add `captchaToken?: string` directly to `SingleFactorPasswordlessParams` ?

Reply:
If `captchaToken` is optional, it shouldn't be an issue.

Reply:
Not exactly, captcha is optional because it is not turned on for all clients, but when it is, it is required.
Plus I don't want to lie on the type system, and what I did matches exactly the types on ciam-app.

Reply:
Some thoughts here:
- I think it doesn't help that we decided to overload `startPasswordless` for step-up flows; alas that's done now but looking back a separate method with its own parameter type would have been clearer.
- Reusing a base type `SingleFactorPasswordlessParams` for both the start and verify steps also doesn't make sense despite the resemblance.

I propose to take this all the way and properly reflect backend expectations:
```typescript
type StartSingleFactorPasswordlessParams = {
  email?: string
  phoneNumber?: string
  captchaToken?: string
}

export type VerifySingleFactorPasswordlessParams = {
  email?: string
  phoneNumber?: string
  verificationCode: string
}
// back compat
export type VerifyPasswordlessParams = VerifySingleFactorPasswordlessParams

type StepUpPasswordlessParams = {
  authType: 'email' | 'sms'
  stepUp: string
}

export type StartPasswordlessParams = StartSingleFactorPasswordlessParams | StepUpPasswordlessParams
// back compat
export type PasswordlessParams = StartPasswordlessParams
```
- `authType` is really unnecessary and while it isn't documented anywhere, we've been wanting to remove this eventually since we can infer the flow type based on the passed identifier. It is already optional at the backend.
- Separate start/verify parameter types without breaking back compat.
- `captchaToken` is optional from the SDK's perspective because as you mentioned, not all tenants have configured it. I don't think it's relevant to have the "captcha" keyword in the type itself.